src/kakarot/instructions/system_operations.cairo

```diff
@@ -773,29 +774,43 @@ namespace SystemOperations {
             return evm;
         }
 
+        // Operation
         if (evm.message.read_only != FALSE) {
             let (revert_reason_len, revert_reason) = Errors.stateModificationError();
             let evm = EVM.stop(evm, revert_reason_len, revert_reason, Errors.EXCEPTIONAL_HALT);
             return evm;
         }
 
-        // Remove this when https://eips.ethereum.org/EIPS/eip-6780 is validated
-        if (recipient_evm_address == evm.message.address.evm) {
-            tempvar is_recipient_self = TRUE;
-        } else {
-            tempvar is_recipient_self = FALSE;
-        }
-        let recipient_evm_address = (1 - is_recipient_self) * recipient_evm_address;
-
-        let recipient_account = State.get_account(recipient_evm_address);
+        // If the account is doesn't have a registered starknet address, it was created in the same tx
+        // If it does, it could've been registered as part of a previous selfdestruct but left empty. Querying
+        // the nonce clears this case.
+
+        // let is_registered = Account.get_registered_starknet_address(self_account.address.evm);
+        // if (is_registered == FALSE) {
+        //     tempvar range_check_ptr = range_check_ptr;
+        //     tempvar syscall_ptr = syscall_ptr;
+        //     tempvar recipient = 0;  // Tokens will be burnt
+        // } else {
+        //     let (original_nonce) = IContractAccount.get_nonce(self_account.address.starknet);
+        //     let is_not_new = is_not_zero(original_nonce);
+        //     tempvar range_check_ptr = range_check_ptr;
+        //     tempvar syscall_ptr = syscall_ptr;
+        //     tempvar recipient = recipient * is_not_new;
+        // }
+
+        // TODO: this causes problems with ef-tests where a CA is deployed with code but nonce zero.
+        // the workaround is as follows: a `created` field in the account struct that tracks created accounts.
```

Comment:
just put here a comment about the burnt eth, and move the comment about the other way to know if an account is new to the model.cairo